packages/date-time-utilities/src/timeMath/timeMath.ts

```diff
@@ -36,3 +36,64 @@ export const ceilMinuteToIncrement = (date: Date, increments: number) => {
   }
   return result;
 };
+
+/**
+ * Returns a date object from the selected time.
+ * @param showSeconds - If the time picker showed seconds
+ * @param useHour12 - If the time picker uses 12 or 24 hour formatting
+ * @param baseDate - The baseline date to calculate the offset of the selected time
+ * @param selectedTime - A string representing the user selected time
+ * @returns A new date object offset from the baseDate using the selected time.
+ */
+export const getDateFromTimeSelection = (
+  showSeconds: boolean,
+  useHour12: boolean,
+  baseDate: Date,
+  selectedTime: string,
+): Date => {
+  const splitValue = selectedTime.split(':');
```

Comment:
It would be simpler and possible more reliable to do this with a regular expression. Something like `/^(\d\d?):(\d\d):?(\d\d)? ?([ap]m)?/i)` should work--the groups are hours, minutes, seconds, apm. https://regex101.com/r/b0A9If/1

Reply:
Resolved by using regex.